### ISSUE TITLE
Incorrect Radical/Square Root containing Fractions

### ISSUE THREAD

**Issue by fermifunktion** [Stone-Zeng/FiraMath#79]
After building the fonts (like to use the bold version of the font) following "Building from source" (without the `patch` as `scripts/axes.py.diff` is non-existent),
square roots containing fractions are not displayed correctly, if using `lualatex`.

``` latex
\documentclass{article}
\usepackage{fontspec}
\usepackage{unicode-math}
\setsansfont{Fira Sans}
\setmathfont{Fira Math}[version=sans]

\begin{document}
\sffamily
\mathversion{sans}
inline math: $\sqrt{a}$ $\sqrt{\frac{a}{b}}$.

display math:
\begin{displaymath}
	\sqrt{a}\sqrt{\frac{a}{b}}
\end{displaymath}
\end{document}
```
![firamath](https://github.com/firamath/firamath/assets/101714805/e828b811-08d2-46a0-a2a2-6535baae5c99)

**Comment by hvoss49** [comment]
Where is the sense of defining mathversion=sans? Fira is by default sans serif.

```
\documentclass{article}
\usepackage{unicode-math}
\setsansfont{Fira Sans}
\renewcommand\familydefault{\sfdefault}
\setmathfont{Fira Math}

\begin{document}

	inline math: $\sqrt{a}$ $\sqrt{\frac{a}{b}}$.
	
	display math:
	\begin{displaymath}
		\sqrt{a}\sqrt{\frac{a}{b}}
	\end{displaymath}

\end{document}
```
<img width="309" alt="Bildschirmfoto 2023-09-29 um 20 18 00" src="https://github.com/firamath/firamath/assets/12891199/02f9a155-03f7-4bc7-88ad-5b76cb7feceb">

**Comment by apoorvpotnis** [comment]
> Where is the sense of defining mathversion=sans? Fira is by default sans serif.
> 
> ```
> \documentclass{article}
> \usepackage{unicode-math}
> \setsansfont{Fira Sans}
> \renewcommand\familydefault{\sfdefault}
> \setmathfont{Fira Math}
> 
> \begin{document}
> 
> 	inline math: $\sqrt{a}$ $\sqrt{\frac{a}{b}}$.
> 	
> 	display math:
> 	\begin{displaymath}
> 		\sqrt{a}\sqrt{\frac{a}{b}}
> 	\end{displaymath}
> 
> \end{document}
> ```
> 
> <img alt="Bildschirmfoto 2023-09-29 um 20 18 00" width="309" src="https://user-images.githubusercontent.com/12891199/271688119-02f9a155-03f7-4bc7-88ad-5b76cb7feceb.png">

I don't know what OP's reasons are here for `mathversion=sans`, but I can imagine where two or more math fonts are used in a document. For example, serif math for normal text and sans serif in titles and section headers, or in figures.

**Comment by fermifunktion** [comment]
Yes, `mathversion=sans` is a reminiscence of a larger file where I defined other math fonts in version 'sansbold' and 'bold'. 
But again, following Herbert's suggestion and removing the definition of `mathversion`, the problem still exists.

For clearness: I do not use the font `FiraMath-Regular.otf` from Tex Live or the official GitHub source, but use the fonts built from source. Unfortunately, these built fonts show the behavior, and I do not understand the source so well to solve the problem on my own in changing the source code...

**Comment by hvoss49** [comment]
can you provide your math font file for download?

**Comment by fermifunktion** [comment]
This is the file I get after following the building rules (without the `patch`):
[FiraMath-Regular.zip](https://github.com/firamath/firamath/files/12774127/FiraMath-Regular.zip)

**Comment by hvoss49** [comment]
For the symbols at the end of the file it looks completely different to the current FiraMath-Regular.otf from TL2023
It looks like a problem with the tools, but I am not sure.

**Comment by hvoss49** [comment]
I downloaded the source and created the fonts. Only FiraMath-Ultra.otf has no problem with the square root

**Comment by zappete** [comment]
Could this be a duplicate of #67 ?

I suggest this because I noticed OP compiles with lualatex  and seems to imply that the problem is not there in xelatex.
I have confirmed this is the case (lualatex gives the problem, xelatex complies fine) by trying OPs example locally (with firamath fonts from github actions)
